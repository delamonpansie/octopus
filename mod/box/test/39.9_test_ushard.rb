#!/usr/bin/ruby
# coding: utf-8

$: << File.dirname($0)
require '39_test_ushard'

$one_env.meta 'shard 1 create por'
$one_env.meta 'shard 1 obj_space 0 create tree unique string 0'

$one.insert [1,"one"], :shard => 1
$two_env.meta 'shard 1 add_replica two'
sleep 0.1

# switch master
$two_env.meta 'shard 1 master two'
$two_env.env_eval do
  wait_for "readable 00000000000000000002.snap" do
    FileTest.readable? '00000000000000000002.snap'
  end
end

$two_env.meta 'shard 1 add_replica three'
sleep 0.1


$two_env.meta 'shard 1 del_replica one'
sleep 0.1

$two_env.env_eval do

  stop
  puts `./octopus --cat 00000000000000000002.snap`.gsub(/ tm:\d+(\.\d+)? /, ' ')
  puts `./octopus --cat 00000000000000000002.xlog`.gsub(/ tm:\d+(\.\d+)? /, ' ')
  puts `grep META octopus.log`.gsub(/^\d+\.\d+ \d+ /, '')
end

