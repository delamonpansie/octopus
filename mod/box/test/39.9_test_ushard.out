# one.ping()
:pong

# two.ping()
:pong

# one.meta(shard 1 create por)
ok

# one.meta(shard 1 obj_space 0 create tree unique string 0)
ok

# one.insert([1, "one"], {:shard=>1})
1

# two.meta(shard 1 add_replica two)
ok

# two.meta(shard 1 master two)
ok

# two.meta(shard 1 add_replica three)
ok

# two.meta(shard 1 del_replica one)
ok

lsn:2 scn:-1 t:snap/snap_initial ver:0 count:1 flags:0x00000000
lsn:2 shard:1 scn:5 t:snap/shard_create SHARD_CREATE shard_id:1 POR Box count:1 run_crc:0xc19a0fa0 master:two repl:one
lsn:2 shard:1 scn:5 t:snap/usr240 CREATE_OBJECT_SPACE n:0 flags:00000003 cardinalty:0 PK: i:0 min_tuple_cardinality:0 cardinality:1 type:POSTREE unique:1 field0:{index:0 type:STRING sort:ASC}
lsn:2 shard:1 scn:5 t:snap/snap_data n:0 <1:"\x01\x00\x00\x00", "one">
lsn:2 shard:1 scn:5 t:snap/shard_final 
lsn:2 scn:-1 t:snap/snap_final 
lsn:2 shard:1 scn:5 t:wal/shard_alter SHARD_ALTER shard_id:1 POR Box count:1 run_crc:0xc19a0fa0 master:two repl:one
lsn:3 shard:1 scn:6 t:wal/shard_alter SHARD_ALTER shard_id:1 POR Box count:1 run_crc:0x90dfeb36 master:two repl:one repl:three
lsn:4 shard:1 scn:7 t:wal/shard_alter SHARD_ALTER shard_id:1 POR Box count:1 run_crc:0x2f5700ed master:two repl:three
xxx/load shard {1} I> META mode: LOADING => shard_create
xxx/load shard {1} I> META mode: PARTIAL_PROXY, SCN: 4, type: POR, peer: [one, two], proxy_addr: 'one/127.0.0.1:33013' => hot_standby/one/127.0.0.1:33013/init
xxx/remote_hot_standby {1} I> META mode: PARTIAL_PROXY, SCN: 4, type: POR, peer: [one, two], proxy_addr: 'one/127.0.0.1:33013' => hot_standby/127.0.0.1:33013/connect
xxx/load shard {1} I> META mode: PARTIAL_PROXY, SCN: 4, type: POR, peer: [one, two], proxy_addr: 'one/127.0.0.1:33013' => wal_final_row
xxx/remote_hot_standby {1} I> META mode: PARTIAL_PROXY, SCN: 4, type: POR, peer: [one, two], proxy_addr: 'one/127.0.0.1:33013' => hot_standby/127.0.0.1:33013/ok
xxx/remote_hot_standby {1} I> META mode: LOCAL, SCN: 4, type: POR, peer: [two, one] => route update, force notify
xxx/remote_hot_standby {1} I> META mode: LOCAL, SCN: 4, type: POR, peer: [two, one] => primary
xxx/box_worker {1} I> META mode: LOCAL, SCN: 6, type: POR, peer: [two, one, three] => route update, force notify
xxx/box_worker {1} I> META mode: LOCAL, SCN: 7, type: POR, peer: [two, three] => route update, force notify
